apps/studio/src/features/dashboard/components/DeleteResourceModal/DeleteResourceModal.tsx

```diff
@@ -0,0 +1,134 @@
+import { useState } from "react"
+import {
+  HStack,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from "@chakra-ui/react"
+import {
+  Button,
+  Checkbox,
+  ModalCloseButton,
+  useToast,
+} from "@opengovsg/design-system-react"
+import { ResourceType } from "~prisma/generated/generatedEnums"
+
+import { isAllowedToHaveChildren } from "~/utils/resources"
+import { trpc } from "~/utils/trpc"
+import { DeleteCollectionModalState } from "../../types"
+
+const getResourceLabel = (
+  resourceType: ResourceType,
+): Lowercase<Extract<ResourceType, "Page" | "Collection" | "Folder">> => {
+  if (!isAllowedToHaveChildren(resourceType)) {
+    return "page"
+  }
+
+  if (resourceType === ResourceType.Collection) {
+    return "collection"
+  }
+
+  return "folder"
+}
+
+const getWarningText = (resourceType: ResourceType): string => {
+  if (!isAllowedToHaveChildren) {
+    return "This cannot be undone"
+  }
+
+  if (resourceType === ResourceType.Collection) {
+    return "All pages under this collection will be deleted. This cannot be undone."
+  }
+
+  return "All folders and pages under this folder will be deleted. This cannot be undone."
```

Comment:
feels almost same as the string above, can use template literals if possible

Reply:
sticking with this cos i think it's clearer ._.